lib/install/management/create.go

```diff
@@ -120,100 +121,85 @@ func (d *Dispatcher) uploadImages(files map[string]string) error {
-			uploadRetryDecider := func(err error) bool {
-				if err == nil {
-					return false
-				}
+			ul := progresslog.NewUploadLogger(d.op.Infof, baseName, time.Second*3)
+			// need to wait since UploadLogger is asynchronous.
+			defer ul.Wait()
 
-				retryCount--
-				if retryCount < 0 {
-					d.op.Warnf("Attempted upload a total of %d times without success, Upload process failed.", uploadRetryLimit)
-					return false
-				}
-				d.op.Warnf("failed an attempt to upload isos with err (%s), %d retries remain", err.Error(), retryCount)
-				return true
-			}
+			return d.session.Datastore.UploadFile(d.op, image, path.Join(d.vmPathName, key),
+				progresslog.UploadParams(ul))
+		}
 
-			// Build retry config
-			backoffConf := retry.NewBackoffConfig()
-			backoffConf.InitialInterval = uploadInitialInterval
-			backoffConf.MaxInterval = uploadMaxInterval
-			backoffConf.MaxElapsedTime = uploadMaxElapsedTime
-
-			uploadErr := retry.DoWithConfig(operationForRetry, uploadRetryDecider, backoffConf)
-			if uploadErr != nil {
-				finalMessage = fmt.Sprintf("\t\tUpload failed for %q: %s\n", image, uploadErr)
-				if d.force {
-					finalMessage = fmt.Sprintf("%s\t\tContinuing despite failures (due to --force option)\n", finalMessage)
-					finalMessage = fmt.Sprintf("%s\t\tNote: The VCH will not function without %q...", finalMessage, image)
-					results <- errors.New(finalMessage)
-				} else {
-					results <- errors.New(finalMessage)
-				}
+		// counter for retry decider
+		retryCount := uploadRetryLimit
+
+		// decider for our retry, will retry the upload uploadRetryLimit times
+		uploadRetryDecider := func(err error) bool {
+			if err == nil {
+				return false
 			}
-			wg.Done()
-		}(key, image)
-	}
 
-	wg.Wait()
-	close(results)
+			retryCount--
+			if retryCount < 0 {
+				d.op.Warnf("Attempted upload a total of %d times without success, Upload process failed.", uploadRetryLimit)
+				return false
+			}
+			d.op.Warnf("failed an attempt to upload isos with err (%s), %d retries remain", err.Error(), retryCount)
+			return true
+		}
 
-	uploadFailed := false
-	for err := range results {
-		if err != nil {
-			d.op.Error(err)
-			uploadFailed = true
+		// Build retry config
+		backoffConf := retry.NewBackoffConfig()
+		backoffConf.InitialInterval = uploadInitialInterval
+		backoffConf.MaxInterval = uploadMaxInterval
+		backoffConf.MaxElapsedTime = uploadMaxElapsedTime
+
+		uploadErr := retry.DoWithConfig(operationForRetry, uploadRetryDecider, backoffConf)
+		if uploadErr != nil {
+			finalMessage = fmt.Sprintf("\t\tUpload failed for %q: %s\n", image, uploadErr)
+			if d.force {
+				finalMessage = fmt.Sprintf("%s\t\tContinuing despite failures (due to --force option)\n", finalMessage)
+				finalMessage = fmt.Sprintf("%s\t\tNote: The VCH will not function without %q...", finalMessage, image)
+			}
+			d.op.Error(finalMessage)
+			return errors.New("Failed to upload iso images successfully.")
```

Comment:
Nit: including `successfully` in the same sentence as `Failed` may be confusing - `Failed to upload iso images` conveys the same intent IMO.

Reply:
I see this string was present before this change - it'd be nice to fix it here.